### ISSUE TITLE
Add commands for checking for upgrade and upgrading src CLI in place

### ISSUE THREAD

**Issue by ryan-blunden** [sourcegraph/src-cli#180]
As a starting point, I like @slimsag 's suggestion from https://github.com/sourcegraph/src-cli/issues/18#issuecomment-425635219:

```
src update check -> checks for updates, doesn't apply them.
src update -> checks for updates, self-applies them. Will prompt for root/admin privileges.
```

I also liked @felixfbecker's suggestion of having a prompt to let the user know there is an upgrade available (screenshot taken from https://www.npmjs.com/package/update-notifier), but add logic so this is only displayed once per day, not on every invocation.

<img width="462" alt="Screen Shot 2020-04-22 at 9 15 17 AM" src="https://user-images.githubusercontent.com/133014/79924005-1f0c5580-847a-11ea-87c2-a05fa4f94889.png">

**Comment by joshuabezaleel** [comment]
Hi @ryan-blunden , will this be open to public contribution? :slightly_smiling_face: 

I think for `src update check` I can use [google/go-github](https://github.com/google/go-github) to get the latest version available.

I am still not sure for the second one, `src update`, though. What comes in my mind is using Go's `os` and `os/exec` to do some "scripting" for "automatically" download the binary and put it at the appropriate folder like the one in the README's installation steps. What do you think?

Thank you very much in advance! :slightly_smiling_face:

**Comment by ryan-blunden** [comment]
@joshuabezaleel Absolutely! We would love you to take this on. @slimsag What do you think about the suggested approach?

**Comment by joshuabezaleel** [comment]
Hi @slimsag , if it would be okay I want to ask questions on this particular issue.
For `src update check`, how would it be different with the current [`src version`](https://github.com/sourcegraph/src-cli/blob/master/cmd/src/version.go#L31-L40) since the `version` command also output the recommended latest version from `/.api/src-cli/version`? Thank you in advance! :slightly_smiling_face:

**Comment by mrnugget** [comment]
Hey @joshuabezaleel, I think the `src version` command does what `src update check` is supposed to do. But `src version` is newer.

**Comment by burmudar** [comment]
Closing this issue as part of repository maintenance. We are no longer using GitHub issues to track src-cli requests or reports. For future questions or support requests, please contact Sourcegraph Support (mailto:support@sourcegraph.com).
